.github/workflows/Manager-CI.yaml

```diff
@@ -21,10 +24,13 @@ jobs:
       cancel-in-progress: true
     timeout-minutes: 60
     runs-on: ubuntu-latest
+    if: github.repository == '5733d9e2be6485d52ffa08870cabdee0/sandbox'
     name: Build all images and run E2E tests
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+      - name: Build and test
+        if: contains(github.event.pull_request.labels.*.name, 'safe to test')
```

Comment:
I would move it as first step to avoid to checkout the code if the build is not supposed to run. Let's rename `Build and test` as `Precondition check` or something similar

Reply:
Hi @danielezonca , I've refactored it. Please take a second look